### ISSUE TITLE
Set auto geolocation tagging time range

### ISSUE THREAD

**Issue by pluja** [photoprism/photoprism#622]
Let me know if I should change the title, I don't really know how this feature should be called.

So Photoprism has a feature that given a 24h time span, it sets the geotag of the images from that span with the geolocation of other images that were taken the same day.

With this in mind, it would be great to be able to change that time span manually. Say I'm on a road trip and I travel from one town to another on the same day. If only photos from one of the towns are geotagged (or none and I manually put the geotag on some of them) being able to set a different timespan than 24h (say 2h for the geotag of 'Town X' and 3h for 'TownY', or maybe more than 24h for 'City Z') would be really useful as one could set the geotagging of several photos at the same time with much more accuracy.

**Comment by lastzero** [comment]
True. We may add config options for this later. Also an option to limit location estimated to specific cameras.

**Comment by dietriro** [comment]
Hi, I ran into the same issue and wrote a small SQL script for "fixing"/removing the location estimates for all pictures where the reference picture was taken > 3h earlier/later. This is kind of a hacky workaround, since the name of the picture is still based on that location. 

Is there any way to change this 24h parameter already or are there any plans? Happy to contribute here. 

In my opinion it might be also useful to have another boolean parameter that one could set if only images should be taken into account (as reference images) which were taken on during the same day. But having a parameter for the hours, which is used during indexing would already be great!

Thanks in advance!

**Comment by lastzero** [comment]
Nope. Yet another config option to add when we have a little time. Hello complexity, old friend!

Remember, you also need to design a user interface and write documentation. Most things are done as quickly as it looks.

**Comment by lapo-luchini** [comment]
> Hi, I ran into the same issue and wrote a small SQL script for "fixing"/removing the location estimates for all pictures where the reference picture was taken > 3h earlier/later

Can you share?

**Comment by dietriro** [comment]
> > Hi, I ran into the same issue and wrote a small SQL script for "fixing"/removing the location estimates for all pictures where the reference picture was taken > 3h earlier/later
> 
> Can you share?

Hi, sorry for the super late response but I have been searching it now for quite a while. In case anyone is still interested in the script, here you go. You can also add more statements to the `WHERE` part where I check for the date, such as the camera id or the folder where it originated from.


```sql
CREATE PROCEDURE UpdatePlaceEstimates() BEGIN DECLARE bDone INT;
DECLARE target_id INT;
DECLARE target_date DATETIME;
DECLARE other_id INT;
DECLARE other_date DATETIME;
DECLARE reallyDone INT;
DECLARE curs CURSOR FOR 
SELECT 
  id, 
  taken_at 
FROM 
  photos 
WHERE 
  place_src = 'estimate';
DECLARE CONTINUE HANDLER FOR NOT FOUND 
SET 
  bDone = 1;
OPEN curs;
SET 
  reallyDone = 0;
SET 
  bDone = 0;
REPEAT FETCH curs INTO target_id, 
target_date;
IF bDone = 1 THEN 
SET 
  reallyDone = 1;
END IF;
SET 
  other_id = NULL;
SET 
  other_date = NULL;
SELECT 
  p.id, 
  p.taken_at INTO other_id, 
  other_date 
FROM 
  photos p 
WHERE 
  p.place_src = 'meta' 
  and p.taken_at >= DATE_ADD(target_date, INTERVAL -4 HOUR) 
  and p.taken_at <= DATE_ADD(target_date, INTERVAL 4 HOUR) 
LIMIT 
  1;
SELECT 
  target_id, 
  target_date, 
  other_id, 
  other_date;
SET 
  bDone = 0;
IF (other_id IS NULL) THEN 
UPDATE 
  photos 
SET 
  photo_country = 'zz', 
  place_id = 'zz', 
  place_src = '', 
  photo_altitude = 0, 
  photo_lat = 0, 
  photo_lng = 0 
WHERE 
  id = target_id;
END IF;
UNTIL reallyDone END REPEAT;
CLOSE curs;
END //
```
